src/linker/Linker.Steps/LinkAttributesStep.cs

```diff
@@ -36,9 +36,17 @@ IEnumerable<CustomAttribute> ProcessAttributes (XPathNavigator nav, ICustomAttri
 				TypeDefinition attributeType;
 
 				string internalAttribute = GetAttribute (iterator.Current, "internal");
-				if (internalAttribute == "RemoveAttributeInstances" && provider != null && !Annotations.IsMarked (provider)) {
-					IEnumerable<Attribute> removeAttributeInstance = new List<Attribute> { new RemoveAttributeInstancesAttribute () };
-					Context.CustomAttributes.AddInternalAttributes (provider, removeAttributeInstance);
+				if (internalAttribute != String.Empty) {
+					if (internalAttribute == "RemoveAttributeInstances") {
+						if (provider != null) {
+							if (!Annotations.IsMarked (provider)) {
+								IEnumerable<Attribute> removeAttributeInstance = new List<Attribute> { new RemoveAttributeInstancesAttribute () };
+								Context.CustomAttributes.AddInternalAttributes (provider, removeAttributeInstance);
+							}
+						} else
+							Context.LogWarning ($"Internal attribute 'RemoveAttributeInstances' can only be used on a type, but is being used on '<{nav.Name}>'", 2048, _xmlDocumentLocation);
```

Comment:
Why is this not error?

Reply:
I think we should keep this a warning - mainly because there's no way to suppress an error (and if this comes from a 3rd party component, there's no way to avoid the error if it can't be suppressed).